### PR TITLE
Use l5d-dst-concrete as request name

### PR DESCRIPTION
**Pull request by adleong**
If linkerd receives a request with the l5d-dst-concrete header set, use that as the request's name.

This can be useful in setups where a request passes through multiple linkerds (such as linkerd-to-linkerd) and the dst assigned by the first linkerd should be used by the second.

### CODE REVIEW

File: router/http/src/main/scala/io/buoyant/router/http/MethodAndHostIdentifier.scala [hunk 1]
```diff
@@ -24,21 +24,23 @@ case class MethodAndHostIdentifier(
...
-    case http.Version.Http10 =>
-      Future.value(
-        (mkPath(Path.Utf8("1.0", req.method.toString) ++ suffix(req)), req)
-      )
-
-    case http.Version.Http11 =>
-      req.host match {
-        case Some(host) if host.nonEmpty =>
+  def apply(req: http.Request): Future[(Dst, http.Request)] =
+    Identifier.dstConcrete(req, baseDtab).getOrElse {
```
Comment (olix0r) [inline router/http/src/main/scala/io/buoyant/router/http/MethodAndHostIdentifier.scala]: This seems a little surprising to me. If I specify a prefix on the identifier, I'd expect this prefix to apply to all names provided by the identifier.

File: router/http/src/main/scala/io/buoyant/router/http/Identifier.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.http.Request
+import com.twitter.finagle.{Dtab, Path}
+import com.twitter.util.{Future, Try}
+
+object Identifier {
+  def dstConcrete(req: Request, baseDtab: () => Dtab): Option[Future[(Dst, Request)]] = {
+    req.headerMap.get("l5d-dst-concrete").flatMap { dst =>
```
Comment (olix0r) [inline router/http/src/main/scala/io/buoyant/router/http/Identifier.scala]: but this isn't actually concrete, is it? It's going to be applied through a dtab.

Comment (olix0r) [inline router/http/src/main/scala/io/buoyant/router/http/Identifier.scala]: Also, we should clear this header if we read it.

**Comment by olix0r** [comment]
It's definitely my preference that we not add this behavior to the existing identifiers -- it makes it harder to reason about them, as it allows for the production of highly irregular names, it's unclear what `consume` means, etc...

I'd prefer that we be able to express composed identifiers like:

```
identifier:
  kind: list
  identifiers:
  - kind: header
    header: my-special-header # override default
    prefix: /h
  - kind: path
    prefix: /p
```

This allows the identifiers to stay to doing one thing well.

**Comment by adleong** [comment]
Closed in favor of https://github.com/BuoyantIO/linkerd/pull/638

**Pull request closed without merging**